app/models/manageiq/providers/openstack/network_manager/refresh_parser.rb

```diff
@@ -64,23 +64,43 @@ def parent_manager_find_device_connected_to_network_port(ems_ref)
     end
 
     def security_groups
-      @security_groups ||= @network_service.handled_list(:security_groups)
+      if @ems.kind_of?(ManageIQ::Providers::Openstack::CloudManager) && ::Settings.ems.ems_openstack.refresh.is_admin
+        @security_groups ||= @network_service.handled_list(:security_groups, {}, true)
+      else
+        @security_groups ||= @network_service.handled_list(:security_groups)
+      end
     end
 
     def networks
-      @networks ||= @network_service.handled_list(:networks)
+      if @ems.kind_of?(ManageIQ::Providers::Openstack::CloudManager) && ::Settings.ems.ems_openstack.refresh.is_admin
+        @networks ||= @network_service.handled_list(:networks, {}, true)
+      else
+        @networks ||= @network_service.handled_list(:networks)
+      end
     end
 
     def network_ports
-      @network_ports ||= @network_service.handled_list(:ports)
+      if @ems.kind_of?(ManageIQ::Providers::Openstack::CloudManager) && ::Settings.ems.ems_openstack.refresh.is_admin
+        @network_ports ||= @network_service.handled_list(:ports, {}, true)
+      else
+        @network_ports ||= @network_service.handled_list(:ports)
+      end
     end
 
     def network_routers
-      @network_routers ||= @network_service.handled_list(:routers)
+      if @ems.kind_of?(ManageIQ::Providers::Openstack::CloudManager) && ::Settings.ems.ems_openstack.refresh.is_admin
+        @network_routers ||= @network_service.handled_list(:routers, {}, true)
+      else
+        @network_routers ||= @network_service.handled_list(:routers)
+      end
     end
 
     def floating_ips
-      @network_service.handled_list(:floating_ips)
+      if @ems.kind_of?(ManageIQ::Providers::Openstack::CloudManager) && ::Settings.ems.ems_openstack.refresh.is_admin
+        @floating_ips ||= @network_service.handled_list(:floating_ips, {}, true)
+      else
+        @floating_ips ||= @network_service.handled_list(:floating_ips)
+      end
```

Comment:
Is it possible to put this code pattern into its own private method, where you pass in :floating_ips or :routers or ... ?